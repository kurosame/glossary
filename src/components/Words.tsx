import Word from '@/components/Word'
import { IWordState } from '@/modules/word'
import GridList from 'material-ui/GridList'
import React from 'react'

interface IProps {
  words: IWordState[]
}

export default class Words extends React.Component<IProps> {
  public render() {
    return (
      <GridList cols={1}>
        {this.props.words.map(w => <Word key={w.id} word={w} />)}
      </GridList>
    )
  }
}
