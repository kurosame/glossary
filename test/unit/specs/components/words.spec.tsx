import Words from '@/components/Words'
import { shallow } from 'enzyme'
import React from 'react'

const words = [
  {
    id: 'Vue.js',
    titles: ['Vue', 'Vue.js'],
    description: 'It a Vue.js'
  }
]

const wrapper = shallow(<Words words={words} />)

test('Match the snapshot', () => {
  expect(wrapper.html()).toMatchSnapshot()
})
