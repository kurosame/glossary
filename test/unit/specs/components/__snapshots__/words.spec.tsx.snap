// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `"<ul class=\\"MuiList-root-1 MuiList-padding-2\\"><li class=\\"MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12\\"><div class=\\"sc-bdVaJa UoSJW\\"><div class=\\"MuiPaper-root-16 MuiPaper-elevation1-19 MuiPaper-rounded-17 MuiCard-root-15\\"><div class=\\"MuiCardHeader-root-43\\" data-test=\\"id\\"><div class=\\"MuiCardHeader-content-46\\"><span class=\\"MuiTypography-root-49 MuiTypography-headline-54 MuiCardHeader-title-47\\"><b>Vue.js</b></span></div></div><div class=\\"MuiCardContent-root-73\\" data-test=\\"titles\\"><div role=\\"button\\" class=\\"MuiChip-root-74\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-79\\">Vue</span></div><div role=\\"button\\" class=\\"MuiChip-root-74\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-79\\">Vue.js</span></div></div><div class=\\"MuiCardContent-root-73\\" data-test=\\"description\\"><div>It a Vue.js</div></div></div></div></li></ul>"`;
